### PR TITLE
Generate cloud resources VPC and OSS

### PR DESCRIPTION
**Pull request by zzxwill**
- Generated cloud resources VPC network and OSS bucket
- Added examples for ProviderConfig and the cloud resources
- Add related docs

### Description of your changes

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # ====================================================================================
 # Setup Project
 
-PROJECT_NAME := provider-jet-template
+PROJECT_NAME := provider-jet-alicloud
```
Comment (turkenh) [inline Makefile:4]: `provider-jet-alicloud` or `provider-jet-alibaba` ?
Should we change repo url if we want to go `provider-jet-alicloud` ?

File: config/oss/bucket/config.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import "github.com/crossplane/terrajet/pkg/config"
+
+// Configure configures individual resources by adding custom ResourceConfigurators.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("alicloud_oss_bucket", func(r *config.Resource) {
+
+		// we need to override the default group that terrajet generated for
+		// this resource, which would be "github"
+		r.ShortGroup = "oss"
```
Comment (turkenh) [inline config/oss/bucket/config.go:11]: fwiw, this custom configuration is probably not needed and the default would already be `oss` in this case different than the tutorial.

File: config/vpc/config.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package vpc
+
+import "github.com/crossplane/terrajet/pkg/config"
+
+// Configure configures individual resources by adding custom ResourceConfigurators.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("alicloud_vpc", func(r *config.Resource) {
+
+		// we need to override the default group that terrajet generated for
+		// this resource, which would be "github"
```
Comment (turkenh) [inline config/vpc/config.go:10]: nit: comment needs to be updated/removed.

File: examples/providerconfig/secret.yaml.tmpl [hunk 4]
```diff
@@ -1,12 +0,0 @@
-apiVersion: v1
```
Comment (turkenh) [inline examples/providerconfig/secret.yaml.tmpl:1]: Is deletion of this file intentional? Don't you want to follow our convention regarding having a template for the expected secret?

**Review by turkenh — APPROVED** [review]
Thanks @zzxwill, looking pretty good, left a couple of minor comments. 

I am seeing `provider-jet-alicloud` everywhere, so guessing we want to rename the repo accordingly?

File: examples/providerconfig/secret.yaml.tmpl (hunk 4 above)
Comment (zzxwill) [reply examples/providerconfig/secret.yaml.tmpl:1]: Yes. As you said, I will keep it.

**Comment by zzxwill** [comment]
@turkenh Thanks a lot for your review. I have addressed all your comments.

File: config/oss/bucket/config.go (hunk 2 above)
Comment (zzxwill) [reply config/oss/bucket/config.go]: Let me keep this line for a while. After removing it, many files are to be removed and more are to be added with `make reviewable`. Please allow me more time to figure it out and I will remove them next time.

**Pull request merged**